packages/google_maps_flutter/lib/src/google_map.dart

```diff
@@ -86,6 +87,30 @@ class GoogleMap extends StatefulWidget {
   /// when the map tries to turn on the My Location layer.
   final bool myLocationEnabled;
 
+  /// The style to be used for the map. The JSON file can be created here at https://mapstyle.withgoogle.com/
```

Comment:
Thanks for this effort to write excellent docs. I think a lot of this can be inferred from the example.

Could you please separate the first line from the rest of the comment per https://www.dartlang.org/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph.

Also, I think it would be useful to cover the following using the Dart docs here:
1. Customize the styling of the base map using a JSON object defined by this string.
2. With style options you can customize the presentation of the standard Google map styles, changing the visual display of features like roads, parks, and other points of interest. As well as changing the style of these features, you can also hide features entirely. This means that you can emphasize particular components of the map or make the map complement the content of your app.
3. The JSON styling file can be created at https://mapstyle.withgoogle.com/
4. Something about what happens when you update the json string.